### ISSUE TITLE
Empty container definition being ignored when default one is specified

### ISSUE THREAD

**Issue by msmootsgi** [nextflow-io/nextflow#691]
As of version 0.29.0 of nextflow, if a default docker container is specified in `nextflow.config` that container is used even if a process has `container null` specified.  In prior versions no container was used if `container null` was specified.  I've attached a small project that illustrates this problem (assuming you're running on Linux and you're not using Ubuntu 18.04).

Here's the differing output:

```
$ NXF_VER=0.28.1 nextflow run main.nf 
N E X T F L O W  ~  version 0.28.1
Launching `main.nf` [high_ride] - revision: 13b975c7e1
[warm up] executor > local
[31/43a2a8] Submitted process > print_default
[ed/43a419] Submitted process > print_local
Local OS: PRETTY_NAME="CentOS Linux 7 (Core)"

Container OS: PRETTY_NAME="Ubuntu 18.04 LTS"

```

and
```
$ NXF_VER=0.29.0 nextflow run main.nf 
N E X T F L O W  ~  version 0.29.0
Launching `main.nf` [stupefied_turing] - revision: 13b975c7e1
[warm up] executor > local
[98/8c4f81] Submitted process > print_default
[33/d83910] Submitted process > print_local
Container OS: PRETTY_NAME="Ubuntu 18.04 LTS"

Local OS: PRETTY_NAME="Ubuntu 18.04 LTS"


```

[nextflow_container_null.tar.gz](https://github.com/nextflow-io/nextflow/files/1985191/nextflow_container_null.tar.gz)

**Comment by pditommaso** [comment]
The issue can be replicated with this script 

```

process print_local {
    container null

    output:
    stdout into p1

    script:
    """
    grep PRETTY_NAME /etc/os-release
    """
}

process print_default {

    output:
    stdout into p2

    script:
    """
    grep PRETTY_NAME /etc/os-release
    """
}

p1.view{ "Local OS: ${it}" }
p2.view{ "Container OS: ${it}" }
```

Config file: 
```
process {
  container = 'ubuntu:latest'
}

docker.enabled = true
```

**Comment by pditommaso** [comment]
Thanks for reporting this. Snapshot `0.29.1-SNAPSHOT` is available for testing.

**Comment by msmootsgi** [comment]
Excellent, the snapshot fixes the problem.  Many thanks!

**Comment by pditommaso** [comment]
Released in version `0.29.1`.
